src/Admin/Translation/Controller/MachineTranslationAdminController.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace App\Admin\Translation\Controller;
+
+use App\Commands\Helpers\CommandHelper;
+use Sonata\AdminBundle\Controller\CRUDController;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\KernelInterface;
+
+abstract class MachineTranslationAdminController extends CRUDController
+{
+  public const TYPE_PROJECT = 'TYPE_PROJECT';
+  public const TYPE_COMMENT = 'TYPE_COMMENT';
+
+  private string $type;
+
+  public function __construct(string $type)
+  {
+    $this->type = $type;
+  }
+
+  public function listAction(Request $request = null): Response
+  {
+    return $this->renderWithExtraParams('Admin/Translation/admin_machine_translation.html.twig', [
+      'action' => 'list',
+      'trimUrl' => $this->admin->generateUrl('trim'),
+    ]);
+  }
+
+  public function trimAction(KernelInterface $kernel): Response
+  {
+    // TODO check permission
```

Comment:
Couldn't figure out how the permission works. For example in `AchievementsAdminController`, I don't know where `ACHIEVEMENT` permissions is defined

```
$this->admin->isGranted('ACHIEVEMENTS')
```

Reply:
It's currently not used anywhere. However, it would allow for more granularity when defining roles that define what can be accessed by whom. Currently, only the SuperAdmin role is working (access to all). Nevertheless, there are plan to fix and extend those role system to allow for specific moderators and so on.

Reply:
I thought roles like `ROLE_CATROWEBADMIN_BLOCK_STATISTICS_PROJECT_MACHINE_TRANSLATION_EDITOR` are working fine in the test environment. Maybe we could allow access to this function if the user has this role?